homework_03/14_martin_damyanov/homework.py

```diff
@@ -0,0 +1,74 @@
+from enum import Enum
+import math
+
+class RelativePosition(Enum):
+    NO_COMMON_POINTS = 1
+    TOUCHING = 2
+    INTERSECTING = 3
+    SAME = 4
+
+def distance(a, b):
+    try:
+        assert isinstance(a, Point)
+        assert isinstance(b, Point)
+    except:
+        print("Given arguments aren't points")
+        exit()
+
+    return math.sqrt(pow(b.x - a.x, 2) + pow(b.y - a.y, 2))
+
+class Point:
+    def __init__(self, x, y):
+        try:
+            assert isinstance(x, float)
+            assert isinstance(y, float)
+        except:
+            print("Coordinates have to be float")
+            exit()
+        
+        self.x = x
+        self.y = y
+
+class Circle:
+    def __init__(self, center, radius):
+        try:
+            assert isinstance(center, Point)
+        except:
+            print("Given argument isn't point")
+            exit()
+
+        try:
+            assert isinstance(radius, float)
+        except:
+            print("The radius has to be float!")
+            exit()
```

Comment:
You can add in the merge request that the radius should be a positive number, right now the radius can be -1 for example and no one will see that this is an invalid input. Spoiler: Before line 36 add assert radius > 0.